### DISCUSSION TITLE
Struggling to install horizon theme

### DISCUSSION THREAD

**Discussion by unknownsouls** [swellstores/community#247] [Help] [answered]
Hi team, 

Wonder if you could help.

I have spent a day trying to solve this, revoked and generated new keys. Deleted and reinstalled node modules, opened a new swell store to check. All to no avail. 

It gets stuck here:

> @swell/horizon@0.4.0 graphql:generate
> graphql-codegen --config codegen.yml -r dotenv/config

✔ Parse Configuration
⚠ Generate outputs
  ❯ Generate to ./graphql.schema.json
    ✖
      Failed to load schema from https://rffs.swell.store/graphql:
      Could not obtain introspection result, received the following as respons…
      { error: { code: "AUTHENTICATION", message: "You must authenticate with …
      GraphQLError: Could not obtain introspection result, received the follow…
      { error: { code: "AUTHENTICATION", message: "You must authenticate with …
      at createGraphQLError (/Users/jane/horizon-2/node_modules/@graphql-tools…
      at getSchemaFromIntrospection (/Users/jane/horizon-2/node_modules/@graph…
      at /Users/jane/horizon-2/node_modules/@graphql-tools/wrap/cjs/introspect…
      at process.processTicksAndRejections (node:internal/process/task_queues:…
      at async UrlLoader.load (/Users/jane/horizon-2/node_modules/@graphql-too…
      at async /Users/jane/horizon-2/node_modules/@graphql-tools/load/cjs/load…
      at async Promise.all (index 5)
      at async loadFile (/Users/jane/horizon-2/node_modules/@graphql-tools/loa…
      at async /Users/jane/horizon-2/node_modules/@graphql-tools/load/cjs/load…
      GraphQL Code Generator supports:
      - ES Modules and CommonJS exports (export as default or named export "sc…
      - Introspection JSON File
      - URL of GraphQL endpoint
      - Multiple files with type definitions (glob expression)
      - String in config file
      Try to use one of above options and run codegen again.
    ◼ Load GraphQL documents
    ◼ Generate
  ❯ Generate to lib/graphql/generated/sdk.ts
    ✖
      Failed to load schema from https://rffs.swell.store/graphql:
      Could not obtain introspection result, received the following as respons…
      { error: { code: "AUTHENTICATION", message: "You must authenticate with …
      GraphQLError: Could not obtain introspection result, received the follow…
      { error: { code: "AUTHENTICATION", message: "You must authenticate with …
      at createGraphQLError (/Users/jane/horizon-2/node_modules/@graphql-tools…
      at getSchemaFromIntrospection (/Users/jane/horizon-2/node_modules/@graph…
      at /Users/jane/horizon-2/node_modules/@graphql-tools/wrap/cjs/introspect…
      at process.processTicksAndRejections (node:internal/process/task_queues:…
      at async UrlLoader.load (/Users/jane/horizon-2/node_modules/@graphql-too…
      at async /Users/jane/horizon-2/node_modules/@graphql-tools/load/cjs/load…
      at async Promise.all (index 5)
      at async loadFile (/Users/jane/horizon-2/node_modules/@graphql-tools/loa…
      at async /Users/jane/horizon-2/node_modules/@graphql-tools/load/cjs/load…
      GraphQL Code Generator supports:
      - ES Modules and CommonJS exports (export as default or named export "sc…
      - Introspection JSON File
      - URL of GraphQL endpoint
      - Multiple files with type definitions (glob expression)
      - String in config file
      Try to use one of above options and run codegen again.
    ◼ Load GraphQL documents
    ◼ Generate
jane@Ellies-MacBook-2 horizon-2 % 

.env info
NEXT_PUBLIC_SWELL_STORE_URL=https://rffs.swell.store
NEXT_PUBLIC_SWELL_PUBLIC_KEY=pk_IisgTcI7SdmQI&&&&&******HBpX
SWELL_STORE_ID=rffs
SWELL_SECRET_KEY=wg53jMxYQ&&&&&******WPqxDHT8m1vw
NEXT_PUBLIC_SWELL_EDITOR=true

Could you please let me know how to get over this?

Likewise, could you please make videos on installation and other practical use of Swell e.g building a schema to make it easier to use on the dashboard and modify as we can on shopify?
Building a simple category page on horizon would be useful, the same goes for integrations. 

Lastly, you have your demo https://swell-horizon-demo.vercel.app/ why not host this on GitHub for us to start with instead of starting from scratch?

Thanks

**Comment by unknownsouls** [comment]
Any help with this?

**Reply by ericingram** [reply]
An updated CLI and guide should be ready early next week

**Reply by logeshswell** [reply, marked as answer]
Your horizon store is available now, can you try `npm install` followed by `npm run dev`?

<img width="1511" alt="Screenshot 2023-05-25 at 1 10 50 AM" src="https://github.com/swellstores/community/assets/100766131/56320a30-51a1-4a5a-8632-deabb91cd093">

**Reply by logeshswell** [reply]
@unknownsouls

**Reply by unknownsouls** [reply]
Hi @logeshswell it didn't work, same graphql error. 

`> @swell/horizon@0.4.0 graphql:generate
> graphql-codegen --config codegen.yml -r dotenv/config

✔ Parse Configuration
⚠ Generate outputs
  ❯ Generate to ./graphql.schema.json
    ✖
      Failed to load schema from https://flon.swell.store/graphql:
      Could not obtain introspection result, received the following as response;
      { error: { code: "AUTHENTICATION", message: "You must authenticate with your public key. Also, make …
      GraphQLError: Could not obtain introspection result, received the following as response;
      { error: { code: "AUTHENTICATION", message: "You must authenticate with your public key. Also, make …
      at createGraphQLError (/Users/dayo/Desktop/horizon-1/node_modules/@graphql-tools/wrap/node_modules/@…
      at getSchemaFromIntrospection (/Users/dayo/Desktop/horizon-1/node_modules/@graphql-tools/wrap/cjs/in…
      at /Users/dayo/Desktop/horizon-1/node_modules/@graphql-tools/wrap/cjs/introspect.js:36:32
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async UrlLoader.load (/Users/dayo/Desktop/horizon-1/node_modules/@graphql-tools/url-loader/cjs/in…
      at async /Users/dayo/Desktop/horizon-1/node_modules/@graphql-tools/load/cjs/load-typedefs/load-file.…
      at async Promise.all (index 5)
      at async loadFile (/Users/dayo/Desktop/horizon-1/node_modules/@graphql-tools/load/cjs/load-typedefs/…
      at async /Users/dayo/Desktop/horizon-1/node_modules/@graphql-tools/load/cjs/load-typedefs/collect-so…
      GraphQL Code Generator supports:
      - ES Modules and CommonJS exports (export as default or named export "schema")
      - Introspection JSON File
      - URL of GraphQL endpoint
      - Multiple files with type definitions (glob expression)
      - String in config file
      Try to use one of above options and run codegen again.
    ◼ Load GraphQL documents
    ◼ Generate
  ❯ Generate to lib/graphql/generated/sdk.ts
    ✖
      Failed to load schema from https://flon.swell.store/graphql:
      Could not obtain introspection result, received the following as response;
      { error: { code: "AUTHENTICATION", message: "You must authenticate with your public key. Also, make …
      GraphQLError: Could not obtain introspection result, received the following as response;
      { error: { code: "AUTHENTICATION", message: "You must authenticate with your public key. Also, make …
      at createGraphQLError (/Users/dayo/Desktop/horizon-1/node_modules/@graphql-tools/wrap/node_modules/@…
      at getSchemaFromIntrospection (/Users/dayo/Desktop/horizon-1/node_modules/@graphql-tools/wrap/cjs/in…
      at /Users/dayo/Desktop/horizon-1/node_modules/@graphql-tools/wrap/cjs/introspect.js:36:32
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async UrlLoader.load (/Users/dayo/Desktop/horizon-1/node_modules/@graphql-tools/url-loader/cjs/in…
      at async /Users/dayo/Desktop/horizon-1/node_modules/@graphql-tools/load/cjs/load-typedefs/load-file.…
      at async Promise.all (index 5)
      at async loadFile (/Users/dayo/Desktop/horizon-1/node_modules/@graphql-tools/load/cjs/load-typedefs/…
      at async /Users/dayo/Desktop/horizon-1/node_modules/@graphql-tools/load/cjs/load-typedefs/collect-so…
      GraphQL Code Generator supports:
      - ES Modules and CommonJS exports (export as default or named export "schema")
      - Introspection JSON File
      - URL of GraphQL endpoint
      - Multiple files with type definitions (glob expression)
      - String in config file
      Try to use one of above options and run codegen again.
    ◼ Load GraphQL documents
    ◼ Generate
npm ERR! code 1`

The graphql is not creating the codegen.yml file, I did a work around to test it out by adding a codegen.yml file to authenticate. See code below, updated the public key. Probably not an ideal solution, do let me know when everything is sorted. Thanks



`overwrite: true
schema: ${NEXT_PUBLIC_SWELL_STORE_URL}/graphql
headers:
  Authorization: "Bearer ${NEXT_PUBLIC_SWELL_PUBLIC_KEY}"
  cache-control: 'no-cache'
documents: 'lib/graphql/**/*.gql'
generates:
  ./graphql.schema.json:
    plugins:
      - 'introspection'
  lib/graphql/generated/sdk.ts:
    config:
      gqlImport: graphql-request#gql
      useTypeImports: true
      rawRequest: true
      documentMode: string
    plugins:
      - 'typescript'
      - 'typescript-operations'
      - 'typescript-graphql-request'

`

**Reply by logeshswell** [reply]
@unknownsouls I fixed the configuration in https://rffs.swell.store and not in store: `flon`
Can you check and confirm if the `rffs` store is working fine?

1. Clone the repo and switch to [swell-cli-sdv2](https://github.com/swellstores/horizon/tree/fix/swell-cli-sdv2) branch.
2. Run the below commands:
```sh
nvm install
nvm use
npm install
npm run dev
```

Please make sure that, 
* You use the public key of Storefront type i.e (`pk_Hn*****FjF8`)
* New env `SWELL_STOREFRONT_ID` is included.

**Reply by unknownsouls** [reply]
Works perfectly. Thank you!
